update full node and telemetry docs for launch of Moonriver (#111)
PureStake/moonbeam-docs: node-operators/networks/compile-binary.md

---
title: Compile the Binary
description: How to compile the Moonbeam binary to run a full Parachain node, gain access to RPC endpoints, and produce blocks, for the Moonbeam Network.
---

# Compile the Moonbeam Binary

![Full Node Moonbeam Banner](/images/fullnode/compile-binary-banner.png)

## Introduction

> There are several ways to get started running a full node on the Moonbeam network. This guide goes through the process of compiling the Moonbeam binary to be run as a systemd service. For a more general overview of running nodes, or to get started with Docker, check out the [Run a Node](/node-operators/networks/full-node) page of our documentation.

Reviewer: ```suggestion
There are several ways to get started running a full node on the Moonbeam network. This guide goes through the process of compiling the Moonbeam binary from Rust source code. For a more general overview of running nodes, or to get started with Docker, check out the [Run a Node](/node-operators/networks/full-node) page of our documentation.
```

Compiling locally does allow the systemd process, but also allow much more. And also works on OSes that don't use systemd.

> This guide is meant for people with experience running [Substrate](https://substrate.dev/) based chains. Running a parachain is similar to running a Substrate node with a few differences. A Substrate parachain node will run two processes: one to sync the relay chain and one to sync the parachain. As such, many things are doubled, for example, the database directory, the ports used, the log lines, and more.

Reviewer: This seems more relevant for the running a node page than the compiling page.

Reviewer: Although the Substrate part is definitely relevant...

```suggestion
This guide is meant for people with experience compiling [Substrate](https://substrate.dev/) based blockchain nodes. A parachain node is similar to a typical Substrate node, but there are some differences. A Substrate parachain node will is a bigger build because it contains code to run the parachain itself, as well as code to sync the relay chain, and facilitate communication between the two. As such, this build is quite large and may take over 30 min and 32GB memory.
```

## Compiling the Binary

The following commands will build the latest release of the Moonbeam parachain.

First, let's start by cloning the moonbeam repo.

```
git clone https://github.com/PureStake/moonbeam
cd moonbeam
```

Let's check out the latest release:

```
git checkout tags/$(git tag | tail -1)
```

> Next, install Substrate and all its prerequisites, including Rust, by executing:

Reviewer: ```suggestion
Next, install a Substrate development environment, including Rust, by executing:
```

For more information and platform specific instructions see https://substrate.dev/docs/en/knowledgebase/getting-started/
